mvcc/kvstore.go

```diff
@@ -271,9 +271,13 @@ func (s *store) Compact(rev int64) (<-chan struct{}, error) {
 			return
 		}
 		close(ch)
-		s.keepMu.Lock()
-		s.keep = nil
-		s.keepMu.Unlock()
+		s.revMu.RLock()
+		if rev == s.compactMainRev {
+			s.keepMu.Lock()
+			s.keep = nil
```

Comment:
I am thinking about appending keep of this compaction to the keep of last compaction if last compaction is still in progress, and only the completion of the very last compaction sets keep to nil.